### PR TITLE
Integrate latest 125x menu ntupler on top of L1 IB v56

### PR DESCRIPTION
**Pull request by artlbv**
#### PR description:

As title reads, this is integrating the Phase-2 menu ntupler changes done for 125x on top of the L1 IB tag v56.
I made this PR since I anyway had to do that rebase* and then we wanted to integrated the changes into the IB. So I let @epalencia @aloeliger @mcepeda decide when/how to merge this.

*What I did was:
```
git cms-merge-topic -u cms-l1t-offline:l1t-phase2-v56
git cms-rebase-topic artlbv:from-CMSSW_12_5_2_patch1
```
And the resolved some of the conflicts. Note that strangely most of these conflicts where trivial since it was just some text reformatting – is it understood why (cms) git cannot handle these automatically?

I know that @EmyrClement must have done something similar to integrate the b-jet score, so we might just use his branch once validated. 

#### PR validation:

NTR here

### CODE REVIEW

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

Attempts to compile this PR succeeded!

|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`cmsenv && scram b -j 8`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found no files with code check issues!

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found 4 files that did not meet formatting requirements:
 - L1Trigger/L1TNtuples/interface/L1AnalysisGeneratorDataFormat.h
 - L1Trigger/L1TNtuples/interface/L1AnalysisPhaseIIStep1DataFormat.h
 - L1Trigger/L1TNtuples/plugins/L1GenTreeProducer.cc
 - L1Trigger/L1TNtuples/src/L1AnalysisPhaseIIStep1.cc

Please run `scram b code-format` to auto-apply code formatting

**Comment by aloeliger** [comment]
@artlbv Could you fix the code formatting on this PR please?

File: L1Trigger/L1TNtuples/interface/L1AnalysisGeneratorDataFormat.h [hunk 1]
```diff
@@ -82,3 +82,4 @@ namespace L1Analysis {
   };
 }  // namespace L1Analysis
 #endif
+
```
Comment (aloeliger) [inline L1Trigger/L1TNtuples/interface/L1AnalysisGeneratorDataFormat.h]: Could you please also get rid of just newline/tab/spaces commits?

**Comment by aloeliger** [comment]
@artlbv Pinging you again on the format fix and removing the newline. Is this ready to go out of draft?

**Comment by artlbv** [comment]
hi @aloeliger thanks for reminding, I was actually going to do that tonight anyway :)
Hope it's fine now!

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

Attempts to compile this PR succeeded!

|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`cmsenv && scram b -j 8`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found no issues with the code checks!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`cmsenv && scram b -k -j 8 code-checks && scram b -k -j 8 code-checks`|

I found no issues with the headers!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`cmsenv && scram b -k -j 8 check-headers`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found no files with code format issues!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`cmsenv && scram b -k -j 8 code-format-all`|

File: L1Trigger/L1TNtuples/python/l1PhaseIITreeStep1Producer_cfi.py [hunk 2]
```diff
@@ -1,60 +1,61 @@
...
+   tkEMTokenHGC = cms.InputTag("l1tLayer1EG","L1TkEmEE",process_name),
 
-#   muonKalman = cms.InputTag("simKBmtfDigis","BMTF"), # we should remove all of these 
-#   muonOverlap = cms.InputTag("simOmtfDigis","OMTF"),
-#   muonEndcap = cms.InputTag("simEmtfDigis",""),
-#   TkMuonToken = cms.InputTag(""),#"L1TkMuons",""),  # removing this from the run 
+                               #   muonKalman = cms.InputTag("simKBmtfDigis","BMTF",process_name), # we should remove all of these
+                               #   muonOverlap = cms.InputTag("simOmtfDigis","OMTF",process_name),
+                               #   muonEndcap = cms.InputTag("simEmtfDigis","",process_name),
+                               #   TkMuonToken = cms.InputTag(""),#"L1TkMuons",process_name,""),  # removing this from the run
```
Comment (aloeliger) [inline L1Trigger/L1TNtuples/python/l1PhaseIITreeStep1Producer_cfi.py:19]: Could you remove commented code please?

File: L1Trigger/L1TNtuples/python/l1PhaseIITreeStep1Producer_cfi.py [hunk 3]
```diff
@@ -1,60 +1,61 @@
...
 
    #Global muons
-#   muonToken = cms.untracked.InputTag("simGmtStage2Digis", ""),
-#   TkGlbMuonToken = cms.InputTag(""),#L1TkGlbMuons",""), # removing this from the run 
+                               #   muonToken = cms.untracked.InputTag("simGmtStage2Digis", "",process_name),
+                               #   TkGlbMuonToken = cms.InputTag(""),#L1TkGlbMuons",",process_name"), # removing this from the run
```
Comment (aloeliger) [inline L1Trigger/L1TNtuples/python/l1PhaseIITreeStep1Producer_cfi.py:23]: Could you remove commented code please

File: L1Trigger/L1TNtuples/python/l1PhaseIITreeStep1Producer_cfi.py [hunk 4]
```diff
@@ -1,60 +1,61 @@
...
+   caloJetHTTToken = cms.InputTag("l1tCaloJetHTT","CaloJetHTT",process_name),
+   caloTauToken = cms.InputTag("l1tCaloJet","L1CaloTauCollectionBXV",process_name),
+   L1HPSPFTauToken = cms.InputTag("l1tHPSPFTauProducerPF","",process_name),
 
-   l1PFMet = cms.InputTag("L1MetPfProducer",""), #emulator
+   l1PFMet = cms.InputTag("l1tMETPFProducer","",process_name), #emulator
 
+   #zoPuppi = cms.InputTag(""), # does not exist anymore!
+                               #l1vertextdr = cms.InputTag("VertexProducer",process_name,"l1vertextdr"), #not used anymore - but kept in the loop just to be sure, not filled to ntuples
+                               #l1vertices = cms.InputTag("VertexProducer","l1vertices",process_name), #not used anymore - but kept in the loop just to be sure, not filled to ntuples
```
Comment (aloeliger) [inline L1Trigger/L1TNtuples/python/l1PhaseIITreeStep1Producer_cfi.py:46]: Could you remove commented code please?

File: L1Trigger/L1TNtuples/src/L1AnalysisPhaseIIStep1.cc [hunk 5]
```diff
@@ -594,6 +609,8 @@ void L1Analysis::L1AnalysisPhaseIIStep1::SetTkJetDisplaced(const edm::Handle<l1t
 }
 
 void L1Analysis::L1AnalysisPhaseIIStep1::SetTkMET(const edm::Handle<std::vector<l1t::EtSum> > trackerMET) {
+  //l1extra_.trackerMET = trackerMET->begin()->hwPt() * l1tmetemu::kStepMET;
+  //l1extra_.trackerMETPhi = trackerMET->begin()->hwPhi() * l1tmetemu::kStepMETPhi - M_PI;
```
Comment (aloeliger) [inline L1Trigger/L1TNtuples/src/L1AnalysisPhaseIIStep1.cc:613]: Please remove commented code

File: L1Trigger/L1TNtuples/test/menu_ntuples_125_rerunL1.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+# End of customisation functions
+
+
+# Customisation from command line
+
+process.source.inputCommands = cms.untracked.vstring("keep *", "drop l1tPFJets_*_*_*")
+# Add early deletion of temporary data products to reduce peak memory need
+from Configuration.StandardSequences.earlyDeleteSettings_cff import customiseEarlyDelete
+process = customiseEarlyDelete(process)
+# End adding early deletion
```
Comment (aloeliger) [inline L1Trigger/L1TNtuples/test/menu_ntuples_125_rerunL1.py:155]: Is this file intended for usage in unit testing, or for general CMSSW tests? Or is this a personal testing file?

File: L1Trigger/L1TNtuples/test/step1_ntuples_125.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+# End of customisation functions
+
+
+# Customisation from command line
+
+# Add early deletion of temporary data products to reduce peak memory need
+from Configuration.StandardSequences.earlyDeleteSettings_cff import customiseEarlyDelete
+process = customiseEarlyDelete(process)
+# End adding early deletion
```
Comment (aloeliger) [inline L1Trigger/L1TNtuples/test/step1_ntuples_125.py:151]: Is this file intended for CMSSW testing/ unit testing? Or is this a personal testing file?

**Comment by EmyrClement** [comment]
As discussed in the offline SW meeting, there's another change that's probably worth including here, rather than making another PR.  We should update from 9x9 histogram jets to 9x9 trimmed histogram jet, which means changing [these lines
](https://github.com/cms-l1t-offline/cmssw/blob/1b8a07208b31cc25ec3f17ffc623b11f131c05c2/L1Trigger/L1TNtuples/python/l1PhaseIITreeStep1Producer_cfi.py#L33-L35) from:

```
 l1pfPhase1L1TJetToken  = cms.InputTag("l1tPhase1JetCalibrator9x9" ,   "Phase1L1TJetFromPfCandidates",process_name), #use the 9x9 case
 l1pfPhase1L1TJetMET  = cms.InputTag("l1tPhase1JetProducer9x9" ,   "UncalibratedPhase1L1TJetFromPfCandidatesMET",process_name), #use the 9x9 case
l1pfPhase1L1TJetSums  = cms.InputTag("l1tPhase1JetSumsProducer9x9",  "Sums",process_name), #use the 9x9 case
   ```
   
   to:
   
   ```
 l1pfPhase1L1TJetToken  = cms.InputTag("l1tPhase1JetCalibrator9x9trimmed" ,   "Phase1L1TJetFromPfCandidates",process_name), #use the 9x9trimmed case
   l1pfPhase1L1TJetMET  = cms.InputTag("l1tPhase1JetProducer9x9trimmed" ,   "UncalibratedPhase1L1TJetFromPfCandidatesMET",process_name), #use the 9x9trimmed case
   l1pfPhase1L1TJetSums  = cms.InputTag("l1tPhase1JetSumsProducer9x9trimmed",  "Sums",process_name), #use the 9x9trimmed case
   ```

**Comment by EmyrClement** [comment]
I should have mentioned the performance change is minimal, but the 9x9trimmed correspond to what's tested with the latest firmware.

**Comment by aloeliger** [comment]
@artlbv I wanted to ping you again on this PR. Could you please take a look at the comments, and when you're done we can open this up to central CMSSW.

**Comment by artlbv** [comment]
hi @aloeliger I would propose to close/halt this PR for now, as we would anyway need to wait for most "snapshot3" PRs to be merged first since we access the "v2 algo" collections here. 

Furthermore, I realised the menu ntupler was never integrated into central cmssw and hence many more changes would have to be reviews in a PR to central (and this is code which dates back from `SVN` times!).

In the meantime I could do a PR to the `Snapshot3` branch to keep the state as was used for the annual review somewhere central and not just in my private repo. No need for thorough review there, just merge as is.
What do you think?

**Comment by artlbv** [comment]
Closing for now as superseded by #1114
